homeassistant/components/sensor/rfxtrx.py

```diff
@@ -46,13 +47,16 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
                 if data_type in event.values:
                     data_types = [data_type]
                     break
+        add_devices_callback(sensors)
         for _data_type in data_types:
-            new_sensor = RfxtrxSensor(None, entity_info[ATTR_NAME],
-                                      _data_type, entity_info[ATTR_FIREEVENT])
-            sensors.append(new_sensor)
+            entity_id = generate_entity_id(ENTITY_ID_FORMAT,
+                                           entity_info[ATTR_NAME],
+                                           hass=hass)
+            new_sensor = RfxtrxSensor(None, entity_info[ATTR_NAME], _data_type,
+                                      entity_info[ATTR_FIREEVENT], entity_id)
             sub_sensors[_data_type] = new_sensor
+            add_devices_callback([new_sensor])
```

Comment:
Ok, then we have to find another solution.

List `data_types` is always the same, and comes from the config.
Example:
`data_types = ['Humidity', 'Temperature' ]`
will create two sensors with the same name, eg: `bedroom`
Then the entity_id will be created [async](https://github.com/home-assistant/home-assistant/blob/7f5f4580746ac581687b52ec06f7c968a02bf98e/homeassistant/helpers/entity_component.py#L184), and the sensors will get the entity_ids  `[bedroom, bedroom_2]` or `[bedroom_2, bedroom]`

So I am not sure how to fix this without making a breaking change.